### PR TITLE
WIP: redo ffprobe command

### PR DESCRIPTION
**Pull request by reynoldsnlp**
Still need code review and tests. In order to test it, we may need to generate an mp4 using `ffmpeg` that has the needed attributes.

### CODE REVIEW

**Comment by reynoldsnlp** [comment]
@WorldsEndless Can you review this code?

**Comment by WorldsEndless** [comment]
Will do

**Comment by WorldsEndless** [comment]
can you provide an example of that the shell command and response should look like? I have not used ffmpeg or ffprobe before. I assume you are trying to get some information about the video  file?

**Comment by reynoldsnlp** [comment]
@WorldsEndless I think it's only used for this. Do you know of a simpler, reliable way to get aspect ratio?

**Comment by WorldsEndless** [comment]
String munging is always brittle, over-sensitive and a little unstable code. But it might be okay if there are no plans for use of ffmpeg anywhere but here. Nonetheless, there appear to be good wrappers in both Java at https://github.com/kokorin/Jaffree, which we could use, or easier still Clojure at https://github.com/w33tmaricich/kawa . In any case, while the library of choice can be sourced from the code (clojars or maven), we need to make sure that both the testing setup and the prod server have ffmpeg set up on them (which you may have taken care of already; let me know).

**Comment by reynoldsnlp** [comment]
ffmpeg is already installed on all the servers. I feel like ffmpeg would be the least brittle (aka most future-proof) way to do this, since it is the de facto media tool on *nix OSes. What do you think?

R
________________________________

**Comment by WorldsEndless** [comment]
Yeah, ffmpeg is itself not brittle. That's why those libraries I cited are not often updated anymore; there is no need! It's the string munging that is prone to breaking and hard to upkeep. I had forgotten that we also need to include the ffprobe in the tests, whether string wrangling or not, so I am going to go ahead and use the Kawa library for this. I'll update this branch with the code and updates.

An added benefit is that we'll be poised for future ffmpeg things, too. I know that isn't planned right now, but it seems definitely in the realm of possibility given that we are operating in the video space.

**Comment by reynoldsnlp** [comment]
Sounds good.

**Comment by WorldsEndless** [comment]
@reynoldsnlp It looks like I got the test worked out. This issue does not make the branching decision based on the resolution, though; that is left for future work.

**Comment by reynoldsnlp** [comment]
@WorldsEndless What does "branching decision based on the resolution" mean? This PR needs to fix the [issue with `display_aspect_ratio`](https://github.com/BYU-ODH/y-video-back-end/issues/161).

**Comment by WorldsEndless** [comment]
"redo ffprobe command" is complete. The resolution data is now available and tested, including manual generation of it. Another issue has been created for acting on that information: issues #167 #169 .

File: project.clj [hunk 1]
```diff
@@ -41,10 +41,10 @@
...
                  [nilenso/honeysql-postgres "0.2.6"]
                  [com.taoensso/timbre "6.0.3"]
-
-                 [tupelo "23.07.04"]]
-
-
+                 [tupelo "23.07.04"]
+                 [ffclj "0.1.2"]
+                                        ;[com.phronemophobic/clj-media "2.3"] ;; This is desirable, not merely a wrapper but using the C APIs. But it has system dependencies on libxcb library, which might effect our deployment targets. 
+                 ]
```
Comment (reynoldsnlp) [inline project.clj:47]: Should this comment be deleted?

File: src/clj/y_video_back/routes/service_handlers/handlers/file_handlers.clj [hunk 2]
```diff
@@ -8,34 +8,49 @@
...
+  (let [calc-gcd (fn [a b]
+              (if (zero? b)
+                a
+                (recur b (mod a b))))
+        [x y :as xy] (map (comp #(Integer/parseInt %) str) [w h])
+        gcd (calc-gcd x y)
+        [rx ry :as xy] (map #(/ % gcd) xy)]
+    (if string?
+      (str rx ":" ry)
+      xy)))
```
Comment (reynoldsnlp) [inline src/clj/y_video_back/routes/service_handlers/handlers/file_handlers.clj:28]: I'm going to log this away as the perfect example of how unreadable Clojure is. XD

File: src/clj/y_video_back/routes/service_handlers/handlers/file_handlers.clj [hunk 3]
```diff
@@ -8,34 +8,49 @@
...
+  (let [result (ffc/ffprobe! [:show_format :show_streams file-path])
+        {:keys [width height display_aspect_ratio]} (-> result :streams first)
+        aspect-ratio (if (and (not compose?) display_aspect_ratio)
+                       display_aspect_ratio
+                       (compute-aspect-ratio width height :string))]
+     ;; use display_aspect_ratio if present, else use width:height
+    aspect-ratio))
+
+(defn _file-create
+  "File Creation, including dimension clipping with based on video aspect ratio"
```
Comment (reynoldsnlp) [inline src/clj/y_video_back/routes/service_handlers/handlers/file_handlers.clj:42]: Delete `with`?

**Review by reynoldsnlp — COMMENTED** [review]
Just a couple tiny housekeeping things.

**Review by WorldsEndless — APPROVED** [review]

**Pull request merged**